src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -142,8 +149,10 @@ void ObjectCopyRequest<I>::send_read() {
   }
 
   auto index = *m_read_snaps.begin();
+
   auto& read_op = m_read_ops[index];
-  if (read_op.image_interval.empty()) {
+  io::WriteReadSnapIds write_read_snap_ids{index};
+  if (read_op.image_interval.empty() || (write_read_snap_ids.first <= m_src_snap_id_start)) {
     // nothing written to this object for this snapshot (must be trunc/remove)
```

Comment:
nit: this comment is not exactly correct now

Reply:
Removed.